libtenzir/builtins/connectors/curl.cpp

```diff
@@ -90,6 +91,69 @@ class curl_loader final : public plugin_loader {
   connector_args args_;
 };
 
+template <detail::string_literal Protocol>
+class curl_saver final : public plugin_saver {
+public:
+  curl_saver() = default;
+
+  explicit curl_saver(connector_args args) : args_{std::move(args)} {
+  }
+
+  auto instantiate(operator_control_plane& ctrl, std::optional<printer_info>)
+    -> caf::expected<std::function<void(chunk_ptr)>> override {
+    // We only use a pointer here to be able to move the handle into the lambda
+    // state.
+    auto handle = std::make_shared<curl>();
+    if (auto err = handle->set(args_.options))
+      return err;
+    // The saver gets its input from the upstream operator, so we ignore any
+    // user-provided request body.
+    if (not args_.options.http.body.data.empty()) {
+      const auto* ptr
+        = reinterpret_cast<const char*>(args_.options.http.body.data.data());
+      auto size = args_.options.http.body.data.size();
+      diagnostic::warning("ignoring non-empty HTTP request body")
+        .hint("{}", detail::byte_escape(std::string_view{ptr, size}))
+        .emit(ctrl.diagnostics());
+    }
+    return [&ctrl, args = args_,
+            handle = std::move(handle)](chunk_ptr chunk) mutable {
+      if (!chunk || chunk->size() == 0)
+        return;
+      // TODO: make this smarter and provide the contentt type. We currently
+      // blindly assume application/json.
+      if (auto err = handle->upload(as_bytes(chunk))) {
+        diagnostic::error("failed to upload chunk ({} bytes) to {}",
+                          chunk->size(), args.options.url)
+          .hint("{}", err)
+          .emit(ctrl.diagnostics());
+        return;
+      }
```

Comment:
@dominiklohmann to follow up on our conversation: here's how the connector blindly stuffs its bytes into the HTTP body.

What would suffice to have is the name of the `format` plugin, because then I could build a table, similar to [what the Brim folks do](https://zed.brimdata.io/docs/lake/api#mime-types):

<img width="768" alt="image" src="https://github.com/tenzir/tenzir/assets/53797/7cd1970a-b853-4e2e-9b5a-4b0dff6c705f">

Reply:
Dominik and I had a conversation about this and the solution we identifies is tagging chunks with metadata.